test/unit/trigger.c

```diff
@@ -0,0 +1,202 @@
+#define UNIT_TAP_COMPATIBLE 1
+#include "unit.h"
+#include <stdbool.h>
+#include <assert.h>
+#include "memory.h"
+#include "fiber.h"
+#include "trigger.h"
+
+/* Length of trigger chains to test. */
+#define TEST_LENGTH 5
+#define FUNC_COUNT (TEST_LENGTH + 3)
+
+struct test_trigger {
+	struct trigger base;
+	int id;
+	int target_id;
+};
+
+/* How many times each trigger was run. */
+int was_run[TEST_LENGTH];
+/* Function codes for each trigger. */
+int funcs[TEST_LENGTH];
+struct test_trigger triggers[TEST_LENGTH];
+
+RLIST_HEAD(list_a);
+RLIST_HEAD(list_b);
```

Comment:
Nit: all global variables should be static.

Reply:
Fixed.